scripts/install_third_party_libs.py

```diff
@@ -23,18 +23,37 @@
 import shutil
 import subprocess
 
+EXPLICIT_CHECK_REQUIRED = False
+PIP_NAME = 'pip'
+try:
+    if 'python2' not in subprocess.check_output(['pip', '--version']).lower():
+        EXPLICIT_CHECK_REQUIRED = True
+except OSError:
+    EXPLICIT_CHECK_REQUIRED = True
+
+if EXPLICIT_CHECK_REQUIRED:
+    try:
+        subprocess.check_call(['pip2.7', '--version'])
+        PIP_NAME = 'pip2.7'
+    except OSError:
+        try:
+            subprocess.check_call(['pip2', '--version'])
+            PIP_NAME = 'pip2'
+        except OSError:
+            raise Exception
+
 # These libraries need to be installed before running or importing any script.
 TOOLS_DIR = os.path.join('..', 'oppia_tools')
 # Download and install pyyaml.
 if not os.path.exists(os.path.join(TOOLS_DIR, 'pyyaml-5.1.2')):
     subprocess.check_call([
-        'pip', 'install', 'pyyaml==5.1.2', '--target',
+        PIP_NAME, 'install', 'pyyaml==5.1.2', '--target',
```

Comment:
Actually, instead of this, can you try https://github.com/oppia/oppia/pull/8145#discussion_r355682347? You might be able to completely bypass the try/catch block above.

Reply:
Ya, that's a nice one. In that case we won't need the command line pip anymore, hence we won't need to check for the default pip name. Thus, we will be using `sys.executable` in these blocks:
```
if not os.path.exists(os.path.join('third_party', 'future-0.17.1')):
    subprocess.check_call([
        PIP_NAME, 'install', 'future==0.17.1', '--target',
        os.path.join('third_party', 'future-0.17.1')])
```
This is what you are saying, right?

Reply:
correct.